### ISSUE TITLE
1.7.7 AGG Module Update

### ISSUE THREAD

**Issue by jamesmrollins** [lzim/teampsd#314]
Below are the changes to the AGG model for sprint 1.7.7

1.  All experimental variables are now changed to "gaming" variables, which will allow incremental time advance.

2.  Changed the Service Proportions from Team Data and Appropriate Service Proportions from Supply switch configuration.  Now Select Mode switch selects between Service Proportions from Team Data and an automatic balancing function that happens "under the hood." The user can see what the automatic balancing values are by selecting the "BC" button next to the "Balanced" selection on the Select Mode switch.  The "i" information for "Balanced" is as follows:

"The percentage of patients who complete an intake who start another service, determined by the available capacity to treat those patients.  If any of these percentages is >1, then there is enough supply for patients to enroll in a service multiple times simultaneously."

![image](https://user-images.githubusercontent.com/30132017/52751144-4678b000-2fa3-11e9-80af-4bf83631eef6.png)

3. Added "Engagement Duration" experiment sliders.

![image](https://user-images.githubusercontent.com/30132017/52751357-0534d000-2fa4-11e9-9ee9-bf61550bf241.png)

4. Updated column header on Team Data Table as shown:

![image](https://user-images.githubusercontent.com/30132017/52751381-1a116380-2fa4-11e9-87a0-1c858f68133d.png)

5. Updated main SIM UI as shown:

![image](https://user-images.githubusercontent.com/30132017/52751405-31505100-2fa4-11e9-8e57-acf1117c4e8e.png)

Thank you,
James

**Comment by lzim** [comment]
@jamesmrollins is this MTL 1.7.7 AGG available in TEST?

thanks

**Comment by jamesmrollins** [comment]
Hi @lzim - AGG should be in test this early this week.  When it is in test, I will issue a checklist and instructions.

**Comment by saveth** [comment]
@lzim @jamesmrollins @branscombj 

I'm noticing in the SIM Team Table some differences from the ones I generate, so I want to double-check in order to update the R code. Here's what it looks like on the SIM:

<img width="379" alt="screen shot 2019-02-27 at 4 29 18 pm" src="https://user-images.githubusercontent.com/12439059/53532751-f4f51880-3aac-11e9-9790-f287a3d4e4b3.png">

We've moved the Appt. Supply from the bottom half of the table to the top half. While there's an Appt. Supply for Intake in the top half of the table, the row for Intake in the bottom half of the table is missing. Are we intentionally dropping the Intake row from bottom half of the table?

**Comment by lzim** [comment]
@saveth First check: Are you reviewing in TEST or PROD?

As you may be able to tell above, more changes are coming...@jamesrollins are the MTL 1.7.7 changes reflected in the Master Crosswalk updates you created last week?

We can discuss tomorrow AM - thanks for reviewing this carefully! 

FYI: @TomRust @branscombj @staceypark @holbrooa @dlkibbe

**Comment by saveth** [comment]
@lzim The changes are in both production and test. These changes appeared after the master crosswalk v15, it's  in the current md version of the master crosswalk.

**Comment by jamesmrollins** [comment]
**Version 1.7.7 AGG is in TEST**: Please consult the design summary and review checklist below to and indicate your "Go, No-Go" response. I will add all participants to a test group named "version_1_7_to_1_8_test" and assign individual worlds. I have truncated the checklist to only cover things that are applicable for this sprint.

**Design Summary:** The AGG Module was:

1. Updated with gaming variables to allow for time stepping.
2. Update main SIM UI to reflect the removal of the "Appropriate Service Proportions from Supply."
3. Removed the "Service Proportions Balanced Across Team Variables" sliders from Experiments Section.  The proportions are now automatically provided by the simulations. Updated the "Select Mode" switch and added a BC button that provides the values automatically calculated. 
4. Added "Engagement Duration" sliders to the Experiments Section.  Added Slider icon to the main SIM UI.
5. In Team Data Table, changed "Median Engagement Duration (wks)" to "Engagement Duration (median)(wks)."

**Simulation Run** (James and Tom)

* [ ]     1. Widgets return same values as the client version of Vensim.
* [ ]     2. Charts are consistent with values from the client version of Vensim.
* [ ]     3. (If new experiments are added) New experiments function properly.
* [ ]     4. (If new functions are added) New functions operate as specified.

**Outputs** (if variables or charts are added) (James, Stacey and Savet)

* [ ]     1. The save-experiment naming function properly encodes any new variable in a manner that is understandable.
* [ ]      2. Team Data is properly mapped to ModelParameters file.

**Training and Documentation** (Debbie, Jane, and Stacey)

* [ ]     1. Changes in simulation are noted in training documents, or are in process.
* [ ]     2. Changes are consistent with dialogue and instruction in current video training.
* [ ]     3. Help descriptions are up-to-date and include explanations of new functions or variables.
* [ ]     4. "i" Information is up-to-date, correct and complete.

**Comment by jamesmrollins** [comment]
Hi @saveth , I spoke with @TomRust this morning regarding the Intake data missing in the lower part of the table.  Because Intake neither has an RVI, nor an Engagement Duration, it was excluded from the table.  Also, try using FireFox again.  We did some tweaks last night.  Mine appears to be working, but I am interested in hearing your experience.  Thanks, James

**Comment by saveth** [comment]
Hi @jamesmrollins, thanks for the fixes. I'm able to log in to both PROD and TEST on FireFox this morning. 👍  I'll update the code to remove Intake from the R script. Also, on the TeamTable for Agg, I noticed that "Appt. Supply" is  shortened  and Psy, CC, MM, etc., are added at the end in parenthesis, however your md master crosswalk  has Psy, CC, MM, etc., followed by the full spelling Appointment Supply, e.g. **CC Appointment Supply (median)**. I'm assuming we're going with the latter?

**Comment by jamesmrollins** [comment]
Hi @saveth .  The AGG Team Data Table used to be smaller forcing some abbreviation. Thus, the abbreviation of "Appointment" is just a legacy. I will change the Team Data Table to reflect the full spelling of "Appointment" as we have room for it.

**Comment by saveth** [comment]
@jamesmrollins cook. I'll update the change on my end too. 😄

**Comment by saveth** [comment]
I'm good with AGG module update. 👍 Just finished updating my code for the Team Data Table. If no one else has any issue, you can close this ticket @jamesmrollins .

**Comment by lzim** [comment]
Great @saveth!

@staceypark is going to start setting up an https://mtl.how/osf project for syncing and version control across "master crosswalk" and "i information" we can discuss on our Monday call with @holbrooa 

Have a great weekend :tada:

**Comment by jamesmrollins** [comment]
@lzim @TomRust @staceypark  

Tom requested the following change to improve usability for the AGG experiments:

In AGG and MM, it’s hard to keep track of all the different hours under Appointment Supply.  It’s easy to inadvertently create hours, or deprive the team, if the hours in any experiment don’t sum to the team’s actual total.  
To help users keep track of their total hours, could you add the cool feature we already have for keeping track of the %s in PSY (see pic below) to the supply sliders in AGG and MM?  I think it would help users to immediately see if they have accidentally allocated more than their actual total supply.  It would have to pull the base case values into the sum, so that even if they only change supply for one service, the total is still accurate (ie, the sum function can’t ignore the values that remain at “BC”).

![image](https://user-images.githubusercontent.com/30132017/55594257-bf6acd00-56f3-11e9-982a-16acf46ddd83.png)


The logic would be:
**If** "Psych Appointment Supply" + "EBPsych Appointment Supply" + "CC Appointment Supply" + "MM Appointment Supply" + "Adjuctive Appointment Supply" + "Group Appointment Supply" + "Intake Appointment Supply" > "Total Appointment Supply"
**Then** show alert "You have exceeded supply"
**Else** show sum "Psych Appointment Supply" + "EBPsych Appointment Supply" + "CC Appointment Supply" + "MM Appointment Supply" + "Adjuctive Appointment Supply" + "Group Appointment Supply" + "Intake Appointment Supply"

_Note:  The "Total" field would always show the results of the following:_
"Total Appointment Supply" - ("Psych Appointment Supply" + "EBPsych Appointment Supply" + "CC Appointment Supply" + "MM Appointment Supply" + "Adjuctive Appointment Supply" + "Group Appointment Supply" + "Intake Appointment Supply")

Here is what it could look like:
![image](https://user-images.githubusercontent.com/30132017/55594651-4d938300-56f5-11e9-9f9e-eff761e71ac7.png)

Please acknowledge that this is acceptable - thank you. james

**Comment by dlkibbe** [comment]
This is very cool @TomRust and @jamesmrollins - question - will the red text get "You've Exceeded Supply" get lost in all the red text in the box? Should it be black font to standout or will it move / zoom to call the viewers attention? If a user is on a mobile phone would they have to scroll down to see the message to figure out how come they can't do anything (that is, will the model "freeze" until they change the numbers to equal the supply?)

**Comment by jamesmrollins** [comment]
Hi @dlkibbe - I am following the same design convention as the warning process in the PSY module experiments.  The contrast study of the reds is adequate to support 508 concerns, so I think we are good there. There is no logic to freeze the model, and like the PSY module, it simply warns them. Since the "Total" field is not a part of the model, it is only advisory; therefore, the experimenter can exceed supply if they wanted to explore that "what if?" Not sure design-wise what to do about the phone view concern.  I need to think on that.

**Comment by lzim** [comment]
What a great solution to this problem @tomrust and @jamesrollins!

I agree it should only warn, as some teams will want to experiment with >100% scenarios.

This will make it much clearer, however.


Excellent work !

**Comment by dlkibbe** [comment]
@jamesmrollins responsive design item. When I minimize the AGG model window (to see it on my smaller laptop screen) the Engagement Pattern, Select Mode box size doesn't adjust so Service Proportions from Team Data overlaps text below
![image](https://user-images.githubusercontent.com/31182258/56150260-067f7a80-5f9e-11e9-90aa-0326361581d7.png)

**Comment by dlkibbe** [comment]
@jamesmrollins see screen shot below for AGG. It looks like the vertical dashed line that is supposed to indicate "today" on the chart is not in the correct place on several charts on the Results Dashboard
![image](https://user-images.githubusercontent.com/31182258/56377599-b8a28680-61d8-11e9-8ae0-a84c04004e67.png)

**Comment by jamesmrollins** [comment]
@dlkibbe Got it.  I was able to recreate it. Looks like it is associated with the "Intake" checkbox.  I will report it and get it resolved.

Thank you,
James

**Comment by lzim** [comment]
@jamesmrollins @dlkibbe @branscombj @staceypark @dlounsbu @TomRust 

Today during the team meeting, I could have sworn that Matt's default position for service proportions was "green" for balanced. But, mine was "red" for team data.

Meaning, I was paralleling him, we were both logging in without adjusting sliders and we had different default slider positions.

**Comment by branscombj** [comment]
Yes, I was surprised that Matt's default was at Balanced. But if that's how the bc ran, it would make the flat lines in the results make better sense, I think.

**Comment by branscombj** [comment]
> @jamesmrollins @dlkibbe @branscombj @staceypark @dlounsbu @TomRust
> 
> Today during the team meeting, I could have sworn that Matt's default position for service proportions was "green" for balanced. But, mine was "red" for team data.
> 
> Meaning, I was paralleling him, we were both logging in without adjusting sliders and we had different default slider positions.

Yes, I was surprised that Matt's default was at Balanced. But if that's how the bc ran, it would make the flat lines in the results make better sense, I think.

**Comment by lzim** [comment]
@jamesmrollins @dlkibbe @branscombj @staceypark @dlounsbu @TomRust

Thanks @branscombj I believe I'm bringing up something else for @jamesmrollins to help us with: My default mode for service proportions in my **Individual world** right now in Agg MTL 1.8 PROD is red for Red in from Team Data, not balanced...When I log in to the **team world** in Agg MTL 1.8 PROD default is green for balanced....

Since I had prepped in my **individual** world, I was really surprised to see the Green Balanced Service Proportions as the default mode in the **team **world. Obviously, they should 1) both be Red for Team Data, and 2) always consistent with each other (not different defaults/basecase for individual versus team).

**_INDIVIDUAL WORLD_**
![team_data_agg](https://user-images.githubusercontent.com/7454688/56391966-02916980-61e5-11e9-94c9-1a91da3c4525.png)

**_TEAM WORLD_**
![balanced_team_world](https://user-images.githubusercontent.com/7454688/56392257-cf9ba580-61e5-11e9-8f36-7d1004c8c9ac.png)

**Comment by lzim** [comment]
@jamesmrollins @dlkibbe @branscombj @staceypark @dlounsbu @TomRust

Note the redundant "actual hours available for service" panels in the "compare services" mode on the bottom left and right.  I also question whether the most effective three panels for the top row of outputs should be the intake variables listed now, regardless of what service is selected in the main section. Proposal to use:

(Appointment Units)
- Actual hours available for service
- Actual return to clinic visit interval

Bottom (Patient Units)
- Patients waiting to start a service
- Patient start rate
- Patients in service
- Patients ending service

![results_dashboard](https://user-images.githubusercontent.com/7454688/56392544-8c8e0200-61e6-11e9-9d72-d866a3b65f1e.png)

**Comment by jamesmrollins** [comment]
@lzim @branscombj @holbrooa @saveth @TomRust @dlkibbe   

Regarding the chart defaults - just let me know what you decide and I will make it happen.

Regarding the default switch position - it is red for Team Data.  That was changed two weeks ago.  Since the simulation uses the same source for code, I cannot explain this occurrence shown above.  I confirmed this with Forio - it should be impossible.  Has anyone tried to recreate it?  I have logged into several individual and team worlds, and it is default red for me.

Regarding the responsive design -  I will report it and see if we can fix it.

Thanks everyone for your patience.

James

**Comment by jamesmrollins** [comment]
@dlkibbe  Can you tell me the resolution of your screen please?  To check, just right click on the laptop's desktop and select "Display Settings." It should bring up a "Display" dialogue.  Scroll down a little and you should see a "Scale and layout" section.  If you could clip that and post it here, that would be great.  See example below:

![image](https://user-images.githubusercontent.com/30132017/56431917-69179580-6280-11e9-8dc8-99c6d50c4879.png)

**Comment by lzim** [comment]
@lzim @branscombj @holbrooa @saveth @TomRust @dlkibbe

> Regarding the chart defaults - just let me know what you decide and I will make it happen.

Everyone: 
- please weigh in ASAP if you have an disagreement with my proposed default charts
- otherwise, @jamesmrollins please follow my recommended panels above for the **Compare Services** Control Panel view

1. Actual hours available for service
2. Actual return to clinic visit interval
3. Patients waiting to start a service
4. Patient start rate
5. Patients in service
6. Patients ending service

> Regarding the default switch position - it is red for Team Data. That was changed two weeks ago.
- Yes, correct, it is red for Team Data as the default Base Case mode.

> Since the simulation uses the same source for code, I cannot explain this occurrence shown above. I confirmed this with Forio - it should be impossible. Has anyone tried to recreate it? I have logged into several individual and team worlds, and it is default red for me.

- Glad it looks okay in other team worlds. In the West Lafayette Team World in question, where we observed this live with our learner team, it looks like the bc_2019_04_16 run was with the Red Team Data default, then the bc_2019_04_18 was run with the Green Service Proportions. 

**_So, did the Team lead adjust it live without any of noticing?_**

**Comment by lzim** [comment]
@lzim @branscombj @holbrooa @saveth @TomRust @dlkibbe

I just reset **West Lafayette TAS Team** it went back to a Red from Team Data default, so my leading hypothesis for what happened is guided by this question:

> So, did the Team lead adjust it live without any of noticing?

**Comment by lzim** [comment]
@TomRust @jamesmrollins @branscombj @dlkibbe @dlounsbu @staceypark 

Do we have a plan for the "bc" save dialogue when there is an option to run a base case with "Team Data" OR "Balanced"? The "bc" mean very different things and we need a way to name those bc runs that differentiates them.

**Comment by lzim** [comment]
@TomRust @jamesmrollins @branscombj @dlkibbe @dlounsbu @staceypark

I just ran a test to see what would happen. We have more issues than I raised in my comment above:

**1. The sim let me save two bc runs with identical names:**
![bc_two_runs](https://user-images.githubusercontent.com/7454688/56448855-d6024e00-62c7-11e9-9188-66fe47624147.png)

**2. When I pull prior runs back up in the experiments section, I **bolds** as if I changed the "Engagement Duration (Group)" Rather than the overall Engagement Pattern Mode, which is not on the list.**
![team_data_on](https://user-images.githubusercontent.com/7454688/56448917-7c4e5380-62c8-11e9-8479-be50ab4b19db.png)

**3. It does the same thing whether the slider has adjusted to "Balanced" or "Team Data" for the Mode: When I pull prior runs back up in the experiments section, I **bolds** as if I changed the "Engagement Duration (Group)" Rather than the overall Engagement Pattern Mode, which is not on the list.**
![balanced_on](https://user-images.githubusercontent.com/7454688/56448937-bae40e00-62c8-11e9-91fe-2d35ad2b3d4d.png)

**4. The next two screen shots show that when I reset to previous experimental values and include the text, the slider is resetting in the opposite direction of the true selection for each bc run. I further confirmed this in my last two notes below, where the question text "What will happen if I run the base case with the green balanced setting on?" produces the red Alt 1 oscillating behavior.** 
![text_balanced_slider_team_data](https://user-images.githubusercontent.com/7454688/56449034-8b81d100-62c9-11e9-9822-244ce4bc7950.png)
![text_team_data_slider_balanced](https://user-images.githubusercontent.com/7454688/56449035-8b81d100-62c9-11e9-9c2c-de7a6cc18dd8.png)

**5. When checking the "?" in the Control Panel, the list of experimental values are identical for both "bc" runs whether I selected Service Proportions from Team Data vs. Balanced. There is not an overall engagement pattern mode in the list. Finally, note in the screen shot I'm seeing two different behaviors, showing I did indeed select two different "service proportion modes for the base case.** 

![two_bc](https://user-images.githubusercontent.com/7454688/56449075-1793f880-62ca-11e9-9e02-d82b10a4cf7e.png)

![bc_values_runs2](https://user-images.githubusercontent.com/7454688/56448824-8c196800-62c7-11e9-8483-fe2d3217d4a4.png)

**6. And, I confirmed that the question text "What will happen if I run the base case with the green balanced setting on?" is associated with the Red Alt 1 bc run that produces oscillation. In contrast, the question text "What happens when I run a base case with Team Data on?" produces the Alt 2 blue flat line:**
![balanced_alt1](https://user-images.githubusercontent.com/7454688/56449245-ba994200-62cb-11e9-86f0-ab98e02139ed.png)
![team_data_alt2](https://user-images.githubusercontent.com/7454688/56449248-be2cc900-62cb-11e9-9b9e-746faf6b9d24.png)

**Comment by jamesmrollins** [comment]
Hi @lzim.  I reran your experiments above.  I could recreate all issues except the save two base cases with the same name (see below).  I got an error message that did not allow me to save. While I am not denying your findings above, I am concerned that we couldn't recreate this problem.  I am going to report it, perhaps DEV has a theory.

![image](https://user-images.githubusercontent.com/30132017/56477980-29a79f80-6460-11e9-8902-6e02508542aa.png)

**Comment by jamesmrollins** [comment]
@lzim I submitted the new chart defaults for Compare Services mode to DEV this evening:

Actual hours available for service
Actual return to clinic visit interval
Patients waiting to start a service
Patient start rate
Patients in service
Patients ending service

**Comment by lzim** [comment]
@jamesrollins

1) did you enter different text and change settings for saving two basecases with the same name?

2) thanks re: compare services default charts

**Comment by lzim** [comment]
@staceypark and @jamesmrollins 

Unfortunately, I am still able to create the same error of saving two different runs in Agg using Safari. No error is coming up for me. 

![re_created_scenario](https://user-images.githubusercontent.com/7454688/56545020-0b5aa600-652b-11e9-9bae-41229e04db14.png)

My Safari details are in the screenshot below, in case they help:
![safari](https://user-images.githubusercontent.com/7454688/56545024-0f86c380-652b-11e9-9568-03bacff4f73c.png)

**Comment by jamesmrollins** [comment]
@lzim I have reported this.  I think it is a Macnominon, as I can't emulate this error in Chrome or Edge.  We will dive into it. - James

**Comment by lzim** [comment]
@jamesmrollins 

I am wondering if it is a Safari 11.1.2 phenomenon - @staceypark did not replicate the issue using Chrome or Safari on her Mac. Right Stacey?

I am still the "special" Safari user who gets weirdness like this snip of with the 280 and the 5600 in Myriad Pro, but the other values in Times New Roman...
![safari_special](https://user-images.githubusercontent.com/7454688/56553166-6350d700-6542-11e9-89ab-4160694ce09d.png)

**Comment by jamesmrollins** [comment]
@dlkibbe  Regarding the Chart rendering issue you reported 6 days ago.  The reason this happened is because you had the Intake checkbox deselected.  Therefore, when the chart went to look for the value and there was none, and there are no other service values to present, it crashed.  We fixed this issue by providing an error message that says, "Chart Not Available" when the intake checkbox is deselected.  To get the chart, just select the Intake checkbox. 

_Note:  @TomRust I just explained this to you on the phone exactly backwards.  Sorry._


 See below for example:

When Intake unchecked:
![image](https://user-images.githubusercontent.com/30132017/56676825-fe959980-6673-11e9-89f6-074e1c1fe555.png)

When Intake checked:
![image](https://user-images.githubusercontent.com/30132017/56676899-208f1c00-6674-11e9-90e8-af8898781402.png)

**Comment by TomRust** [comment]
@jamesmrollins @lzim --
I think I figured out what is up with the Balanced vs Team Data Switch.  It looks like it is "wired" backwards, i.e., the link between the graphic and the number it puts into the model at time zero.  The behavior we should see in Patients Waiting to Start when the switch is set to Team Data is a flat line from week -13 to 0, then oscillations or a drop to almost zero.    When the switch is set to Balanced, then we should get flat lines across all quarters.  

When the team selects "Balanced," the model should receive a "0"
When the team selects "Team Data," the model should receive a "1"

@jamesmrollins -- The switch is designed to start working at time 0, not when the model is initialized, so we don't have the same issue in AGG that we do for MM and CC.

**Comment by branscombj** [comment]
@jamesmrollins @TomRust @lzim In Agg today noticed that Starting a Service Rate chart had units labeled **patients** - should be **patients per week**

Also: RVI Delays Affects [sic] Treatment Decisions does not show as a complete loop (even with all complexity revealed) - needs more arrows. And fix label to **Delays Affect**. Thanks!

**Comment by TomRust** [comment]
@jamesmrollins --
There is another issue, though: 
The widgets controlling the rate and stock levels in the model diagram for Intake are getting weird numbers.  For "Patients Waiting for Intake Evaluation," this number should always be the same as the inflow and outflow rates associated with that stock (as the default wait time for Intake is one week).  When the user has the Intake radio button selected, we shouldn't see any numbers in the rest of the patient flows or stocks.  They should be all displaying zeros.

![image](https://user-images.githubusercontent.com/32961420/56686142-644c4a80-66a1-11e9-8109-237e18508a88.png)

**Comment by jamesmrollins** [comment]
@TomRust I have questions regarding the switch wired backwards issue:

1.  The model is set correctly in Epicenter - it is in default mode, which is Use Team Data on New Pt Referrals = 1:
![image](https://user-images.githubusercontent.com/30132017/56699843-19840000-66ac-11e9-902f-6dcc6e0f2e93.png) 

2.  My assumption based on explanation you provided in Vensim, if the switch above is set to "0" the, "Default will ignore the current state, and instead match patient starts to allocation of clinician hours" or in other words "Balanced" - right?

3.  It appears the model is treating the value from the variable as a constant - meaning, a single value across time.  It does start at -13 and move to 0.  In each step it returns a 1.
![image](https://user-images.githubusercontent.com/30132017/56700056-1b01f800-66ad-11e9-92c0-fc4074b21895.png)

4. Patients Waiting to Start appear to be consistent between the SIM UI model and the client version of Vensim.  What am I missing?

![image](https://user-images.githubusercontent.com/30132017/56700488-569dc180-66af-11e9-98a8-f6291cf3fdf6.png)

**Comment by jamesmrollins** [comment]
@TomRust - Yes, somehow the Display Patient Services radio buttons are confused.  I will look into this tonight and correct.

![image](https://user-images.githubusercontent.com/30132017/56700802-eee87600-66b0-11e9-95cb-364ac9a919b3.png)

**Comment by jamesmrollins** [comment]
@branscombj  We are having chart units-of-measure problems.  We are working to correct.  

Regarding the incomplete loop.  Can you show me what you are talking about?  Below is what I can see:

![image](https://user-images.githubusercontent.com/30132017/56700873-4b4b9580-66b1-11e9-95fe-ced233f993c2.png)


Regarding the "Effect of RVI"  versus the active voice of "RVI Affects," for the sake of consistency, that will require a global change across all the model sensitivity causal loops.  This will take some time.

**Comment by lzim** [comment]
Thanks @jamesmrollins 

@TomRust @branscombj @dlkibbe @dlounsbu @staceypark 

I checked the facilitator dashboard, and no one has run additional runs in their individual world yet, but we'll need to be careful about providing guidance to them if changes would impact their Done/Do work.

Thanks!

**Comment by lzim** [comment]
Hi @jamesmrollins 

I am so sorry you are feeling sick!  I wasn't sure based on the thread whether any of these things moved forward or not. We are planning for an 11:30AM Pacific Time co-Facilitator training tomorrow.

If you are able to update us on the issues above, we welcome when you can.

Feel better 🌡

**Comment by jamesmrollins** [comment]
**Current Situation**
1. The Display Patient Services - widget values disparity is resolved in Test - will promote to PROD tonight.
2. We are working on the Responsive Design issue that Debbie reported.  We didn't have that particular resolution coded, so we have to code it.  In the future we should deliberately decide which resolutions we want to support. 
3. I don't know which loops Jane are missing.  Also, while the language can be improved on the Effects/Sensitivity variables and sliders, I cannot make those changes by tomorrow.  I will incorporate those into the 508 work stream.
4. We are currently stuck on getting the chart labels to work properly in AGG.  I hope we get to the bottom of this by tomorrow. I'm optimistic.
5. The MM Mode switch is also a bugger of a problem that will likely not get solved before next week.  If our current theory holds, we should have a prototype by tomorrow, but then we have to get into the switch redesign after that, which will likely take a day - I will see what I can do to compress this.  It would have been good to have discovered this in March during MM test.
6. If I feel a little better this afternoon, I will see if I can get the gifs that Stacey needs done.
7. I am unclear about Tom's AGG Mode switch design.  I think we have it right, so I need to talk to him.

**Things that probably will not make the launch date next week:**
1. Completed Team PSD website.  If I can get started on this by Sunday, we can at least accomplish the separation of the website from MTL DEMO.  Right now, the issues above are taking priority.
2. I am also short around 20 portraits and bios.  I need those to complete the maps and such.
3. The floating "Partner with Us" button and data collection form.
4. SSO - hopefully IT will get me set up soon, but they don't have a golden track record.

**Comment by lzim** [comment]
Thanks for the update @jamesmrollins We'll check in tomorrow about the AGG switch, in particular :smile:

Rest up and thank you 🛌

**Comment by jamesmrollins** [comment]
@lzim @TomRust  AGG good to go.

**Comment by lzim** [comment]
Thank you @jamesmrollins I hope you're feeling better!

Lindsey

**Comment by staceypark** [comment]
@jamesmrollins We are unable to move green basecase box left to right in AGG, engagement pattern mode. It only moves up and down

**Comment by jamesmrollins** [comment]
@staceypark Noted and on bug report.

**Comment by jamesmrollins** [comment]
@dlkibbe We have built a responsive design to accommodate your screen resolution.  When you have a moment, will you check it and see if it is adequate, please?  Thank you. James

**Comment by jamesmrollins** [comment]
@staceypark The green Base Case Values box should be able to move around the entire browser window now.  Thank you, James

**Comment by staceypark** [comment]
@jamesmrollins Green Base Case values box still only moving up and down, but not left to right (the green BC one in Select Mode) 
![image](https://user-images.githubusercontent.com/31089501/56983900-ab22c000-6b39-11e9-82c5-b924668c9cf2.png)

**Comment by TomRust** [comment]
@jamesmrollins -- 

I don't think the widget that sums up the appointment supply in the experiment slider is working quite right.  

The sum really shouldn't ever go negative.  :)  

![image](https://user-images.githubusercontent.com/32961420/56989788-48451f00-6b61-11e9-8e44-3ee7acf0d5ad.png)

Could the widget always show the sum, even if all sliders are at "BC"?

**Comment by jamesmrollins** [comment]
@TomRust  - yeah, was working before, not sure why it isn't now.  Will investigate.
 
@staceypark - It works, just hasn't been promoted to PROD yet.

**Comment by jamesmrollins** [comment]
@TomRust we didn't get to fixing the experiment slider last night, but will definitely have it done by morning.

**Comment by lzim** [comment]
@jamesmrollins @TomRust and @holbrooa 

I see a hard-coded "i" information in AGG for **engagement duration** missing XXXX in the screenshot below. It should also be re-orded to go after RVI. Do we have this "i" info in OSF already?
![engagement_duration](https://user-images.githubusercontent.com/7454688/57116231-a8c29080-6d08-11e9-821c-916f3a1bab2e.png)

**Comment by jamesmrollins** [comment]
@lzim - Not sure what happened.  I will report it and get it fixed tonight. I will also reorder it to follow RVI.  Also, the Appointment Supply warning is not appearing when sliders exceed TD Appointment Supply.  I have noted this as well.

**Comment by lzim** [comment]
@jamesmrollins Open Science Framework (OSF) at [mtl.how/osf](https://mtl.how/osf)
where the "i" information pop-ups and Master Crosswalk "live." [direct link](https://osf.io/ygp36/)

Looks good for the definition.

**Comment by TomRust** [comment]
@dlkibbe , @holbrooa -- Just following up from my Lucid action items: Any ideas on how to write up a blurb for Adjunct services that doesn't use the phrase "catch-all"?  Even though that is exactly what it is. :smiling_imp: 

maybe something about the variety of what it does include, not simply defining it by what its not?

**Comment by branscombj** [comment]
> 
> maybe something about the variety of what it does include, not simply defining it by what its not?

It really is defined by what it's not, so I don't think you can avoid saying that. But the emphasis can be on the things that we ARE most focused on. There are specific professional services, particularly those for which an evidence-based practice is available, that we want to optimize. We define and count and model those. We know there are numerous other activities that staff perform - from x to y to z (name a few). We put them in the model too because they matter - they take up time and play a part in the system - but we group them together because they are not primary targets of our quality improvement effort.

**Comment by TomRust** [comment]
@holbrooa @dlkibbe @branscombj -- How about this then:

Adjunct Services include numerous other activities the team performs - from _telephone to y to z_. This estimate of hours per week is based on all encounters that are not coded into any of the other service categories. These activities are grouped together because they not the primary focus of Modeling to Learn, but should not be ignored either. They are important to developing a complete understanding of the team and how they are spending their finite patient-facing care time.  

Could ?someone? fill in two other example encounters included in Adjunct?

**Comment by dlkibbe** [comment]
Thanks @TomRust - a question and a suggestion for the last sentence:
1. For the x, y, and z list - Would it be helpful for the healthcare professionals to see reference to specific, common CPT codes that are coded in Adjunct Services in the VA?
2. Last sentence - potential edit (may be too wordy) "Adjunct services are important to developing a comprehensive understanding of how the team is spending their finite patient-facing care time."

**Comment by TomRust** [comment]
Thanks @dlkibbe !  Sounds good to me.

**Comment by lzim** [comment]
@TomRust @dlkibbe and @staceypark 

I would recommend this:

"Adjunct Services include encounters that do not fit into the other listed categories. These services are diverse, and can vary widely across teams, but should not be ignored. They are important to developing a complete understanding of the team and how they are spending their finite patient-facing care time."

**Comment by TomRust** [comment]
@lzim -- love it! concise!

@holbrooa -- I think this explanation for Adjunct (above) goes in the model parameters tab.  Can you drop it in there?

**Comment by lzim** [comment]
@holbrooa Is this adjunct definition in Model Parameters?

> "Adjunct Services include encounters that do not fit into the other listed categories. These services are diverse, and can vary widely across teams, but should not be ignored. They are important for developing a complete understanding of the team and how they are spending their patient-facing care time."

If it has been incorporated in the Master Crosswalk, then @jamesmrollins can this issue be closed?

I'm looking at all the issues under the 1.8 Sprint Milestone and hoping that they can be closed.
Thanks!
🏃‍♀ 

FYI: @staceypark
